python/ray/tune/trial.py

```diff
@@ -346,6 +410,7 @@ def update_last_result(self, result, terminate=False):
         self.last_result = result
         self.last_update_time = time.time()
         self.result_logger.on_result(self.last_result)
+        self.syncer.sync_down(TrialDirSchema.REMOTE_LOGDIR)
```

Comment:
wait ... the syncer is in charge of both results and checkpoints?

Then why is there a separate syncer in the UnifiedLogger?
And also, why specify the source and target in instantiation in the first place then?

Reply:
> Then why is there a separate syncer in the UnifiedLogger?

so actually there's only one syncer. I'm passing in a no-op to `UnifiedLogger`. it makes more sense to me to have the trial control the syncing (since it's aware of the directory structure and logging should be separate from checkpointing, conceptually).

> And also, why specify the source and target in instantiation in the first place then?

yeah you're right i just did it this way since it was less code to change, i can change it to take the full paths at execution.

Reply:
Then we don't need a syncer in the UnifiedLogger anymore right?

Reply:
yeah i left it in since i wasn't sure if other places were using that syncing functionality but it seems like they don't so i just got rid of it